docs/src/guides/pip.md

```diff
@@ -0,0 +1,257 @@
+during build-time, while the libraries should be propagated. We use `config.deps` instead of a conventional `pkg` here to be able to "override" inputs via the [module system](../modules.md).
+6. Tell the [buildPythonPackage module](../reference/buildPythonPackage/index.md) to verify that it can import the given python module from our package after a build.
+7. Tell the [pip module](../reference/pip/index.md) which dependencies to lock using the same syntax as 
+a `requirements.txt` file. Here: `pillow==10.4.0`.
+8. `pip` uses binary wheel files if available by default, we well it not to do so in order to ensure
+a build from source.
+
+### Initialise the repostory
+
+If you use `git`, you need to add `flake.nix` and `default.nix` to your git index so that they get copied to the `/nix/store` and the commands below see them:
+
+```shell-session
+$ git init
+$ git add flake.nix default.nix
+```
+
+### Create a lock file
+
+The next step is to create a lock file by running the packages `lock` attribute. This does a `pip install --dry-run` under the hood and pins the exact packages pip would install.
+
+```shell-session
+$ nix run .#default.lock
+$ git add lock.json
+```
+
+!!! note
+
+    Due to limitations in `pip`s cross-platform support, the resulting
+    lock-files are platform-specific!
+    We therefore recommend setting `paths.lockFile` to `lock.${system}.json`
+    for all projects where you use the pip module.
+
+    Check out the [pdm module](../reference/WIP-python-pdm/index.md) if you need a solution that
+    allows locking for multiple platforms at once!
+
+### Build it
+
+After that's done, we can finally build it:
+
+```shell-session
+$ nix build .#default
+```
+
+Congratulations, you just built your first python package with dream2nix! The resulting package can be used with any other nix python package as long as it uses the same version of python.
+
+
+## Example: Local python project
+
+In our second example, we package are going to package a simple, fictional python package called `my_tool`. 
+Its code and nix expressions are  available in full in [./examples/packages/languages/python-local-development](https://github.com/nix-community/dream2nix/tree/main/examples/packages/languages/python-local-development).
+
+### Code
+
+```nix title="default.nix"
+{
+  config,
+  lib,
+  dream2nix,
+  ...
+}: let
+  pyproject = lib.importTOML (config.mkDerivation.src + /pyproject.toml); # (1)
+in {
+  imports = [
+    dream2nix.modules.dream2nix.pip # (2)
+  ];
+
+  deps = {nixpkgs, ...}: {
+    python = nixpkgs.python3; # (3)
+  };
+
+  inherit (pyproject.project) name version; # (4)
+
+  mkDerivation = {
+    src = lib.cleanSourceWith { # (5)
+      src = lib.cleanSource ./.;
+      filter = name: type:
+        !(builtins.any (x: x) [
+          (lib.hasSuffix ".nix" name)
+          (lib.hasPrefix "." (builtins.baseNameOf name))
+          (lib.hasSuffix "flake.lock" name)
+        ]);
+    };
```

Comment:
I'm not sure if we should include the filtering of the source in this example, since it's not really needed in a "getting started" guide, and most people would probably take this example and copy paste it.

Reply:
Yes, it's a good question. The exaple here is, as you probably noticed, a verbatim copy of the `python-local-development` one in ./examples. If we remove it here, I'd prefer to remove it over there as well for consistency.

On one hand i can see that it opens too many topics at once for people, on the other hand getting an idea of whats copied to the nix store when is important to have any change of understanding why a rebuild happens - even more so with devshells.

I see if i manage to do a better explanation on that, or otherwise remove it for now.